agent/app/agent.go

```diff
@@ -447,10 +447,15 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 	// Register the container instance
 	err = agent.registerContainerInstance(client, vpcSubnetAttributes)
 	if err != nil {
-		if isTransient(err) {
+		if isTerminal(err) {
+			// On unrecoverable error codes, agent should terminally exit.
+			seelog.Warnf("Agent will terminally exit, unable to register container instance: %v", err)
```

Comment:
nit: should use `logger` instead of seelog here?

Reply:
seelog.Warnf is chosen here to conform to existing style of using it for warning messages in this specific file i.e. there were 9 seelog.warn vs 2 logger.warn - so I opted for the more commonly used

There are 9 cases I used as reference found :
1. https://github.com/aws/amazon-ecs-agent/blob/41d593c6110b7dc43bea3db6f871a15591917af0/agent/app/agent.go#L945
2. https://github.com/aws/amazon-ecs-agent/blob/41d593c6110b7dc43bea3db6f871a15591917af0/agent/app/agent.go#L953


Where as logger is also used but mostly for logger.critical and logger.info. There are only 2 cases of logger.Warn: https://github.com/aws/amazon-ecs-agent/blob/41d593c6110b7dc43bea3db6f871a15591917af0/agent/app/agent.go#L1036

Reply:
Got it, thanks for the justification. I am no expert but it seems more idiomatic to conform to the in-house logger used in this library

Reply:
also, I think a log statement for a terminal error should be logged with "Critical" level instead of a "Warn".

Reply:
Ack - implementing recommended